### PR TITLE
feat: GALI-886-BYOB setting

### PR DESCRIPTION
**Pull request by MBtea**
Issue #, if available:

Description of changes:

- Add UI code to disable registering option of read & write study when egress store feature is enabled
- Add backend code to disable registering readwrite study when egress store feature is enabled 
- Add unit test of backend study service

Checklist:

- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jn1119 — DISMISSED** [review]

File: addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js [hunk 1]
```diff
@@ -83,9 +83,13 @@ class DataEgressService extends Service {
     } catch (error) {
       throw this.boom.notFound(`Error in fetch egress store info: ${JSON.stringify(error)}`, true);
     }
+
+    if (egressStoreScanResult.length === 0) {
+      return null;
+    }
```
Comment (jn1119) [inline addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js:89]: Do we need a unit test here?

File: addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js [hunk 2]
```diff
@@ -190,6 +194,13 @@ class DataEgressService extends Service {
     }
 
     const egressStoreInfo = await this.getEgressStoreInfo(environmentId);
+    if (!egressStoreInfo) {
+      await this.audit(requestContext, {
+        action: 'terminated-egress-store',
+        body: 'No egress store found to be terminated',
+      });
+      return null;
```
Comment (jn1119) [inline addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js:202]: Do we need a unit test here?

**Review by jn1119 — DISMISSED** [review]

File: addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js (hunk 2 above)
Comment (MBtea) [reply addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js:202]: tests added in commit: 2a80047064bb961f9e8b9fa14643cb65033161f7

File: addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js (hunk 1 above)
Comment (MBtea) [reply addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js:89]: tests added in commit: 2a80047064bb961f9e8b9fa14643cb65033161f7

**Review by jn1119 — APPROVED** [review]

**Pull request merged**